src/test/clojure/clara/test_rules.clj

```diff
@@ -5025,3 +5029,31 @@
     (is (= (query end-session-equal-facts lousy-weather-query)
            (query end-session-identical-facts lousy-weather-query)
            [{}]))))
+
+(deftest test-alpha-batching-with-multiple-ancestors
+  ;; see issue 257 for details of the bug surronding this test
+  (let [accum-state (atom #{})
+        
+        stateful-accum (acc/accum
+                         {:initial-value []
+                          :reduce-fn conj
+                          :retract-fn (fn [items retracted] (remove #{retracted} items))
+                          :convert-return-fn (fn [items] 
+                                               (do 
+                                                 (swap! accum-state conj items)
+                                                 items))})
+        
+        map-rule (dsl/parse-rule [[?maps <- stateful-accum :from [List]]]
+                                 ;; don't care about whats inserted
+                                 (->LousyWeather))
+        
+        identity-list (ArrayList.)
```

Comment:
This is an ArrayList, not an "identity-list".  I presume this is a remnant of your earlier approach of having this be an IdentityHashMap. :)